contracts/btc-finality/src/finality.rs

```diff
@@ -707,53 +691,67 @@ pub fn compute_active_finality_providers(
     let last_finalized_height = get_last_finalized_height(&deps.as_ref())?;
 
     // Get all finality providers from the staking contract, filtered
-    let mut batch = list_fps_by_power(&cfg.staking, &deps.querier, None, QUERY_LIMIT)?;
+    let mut batch = query_fps_by_total_active_sats(&cfg.staking, &deps.querier, None, QUERY_LIMIT)?;
 
-    let mut finality_providers = vec![];
+    let mut fp_power_table = HashMap::new();
     let mut total_power: u64 = 0;
-    while !batch.is_empty() && finality_providers.len() < max_active_fps {
+    while !batch.is_empty() && fp_power_table.len() < max_active_fps {
         let last = batch.last().cloned();
 
         let (filtered, running_total): (Vec<_>, Vec<_>) = batch
             .into_iter()
             .filter(|fp| {
-                // Filter out FPs with no voting power
-                if fp.power == 0 {
+                // Filter out FPs with no active sats
+                if fp.total_active_sats == 0 {
+                    return false;
+                }
+                // Filter out slashed FPs
+                if fp.slashed {
                     return false;
                 }
                 // Filter out FPs that are jailed.
                 // Error (shouldn't happen) is being mapped to "jailed forever"
-                JAIL.may_load(deps.storage, &fp.btc_pk_hex)
+                if JAIL
+                    .may_load(deps.storage, &fp.btc_pk_hex)
                     .unwrap_or(Some(JAIL_FOREVER))
-                    .is_none()
+                    .is_some()
+                {
+                    return false;
+                }
                 // Filter out FPs that don't have timestamped public randomness
-                && has_timestamped_pub_rand_commit_for_height(&deps.as_ref(), &fp.btc_pk_hex, env.block.height, Some(last_finalized_height))
```

Comment:
Here's the original check for timestamped pub rand commit for height.

Correct me if I'm wrong, but even if a FP was allowed to submit a sig without timestamped pub rand, he would have been excluded from the tallying here.

So, besides some renaming for clarity (which is good) and a bug fix related to filtering out slashed FPs, this PR looks more like an alignment with Babylon than anything else to me (which again, is good).

Reply:
So if the FP has no timestamped pub rand at this height, it won't be included in the voting power table and the finality sig will be rejected due to zero voting power